pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -226,17 +235,19 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 		}
 
 		// If the object is owned by remote config and newer, we need to update the spec in Kubernetes
-		// If Kubernetes is newer, we wait for RC to update the object in our internal store.
+		// If Kubernetes is newer, we update creation timestamp and wait for RC to update the rest of object in our internal store.
 		if podAutoscalerInternal.Spec().RemoteVersion != nil &&
 			podAutoscaler.Spec.RemoteVersion != nil &&
 			*podAutoscalerInternal.Spec().RemoteVersion > *podAutoscaler.Spec.RemoteVersion {
 			err := c.updatePodAutoscalerSpec(ctx, podAutoscalerInternal, podAutoscaler)
 
-			// When doing an external update, we stop and reqeue the object to not have multiple changes at once.
+			// When doing an external update, we stop and requeue the object to not have multiple changes at once.
 			c.store.Unlock(key)
 			return autoscaling.Requeue, err
 		}
 
+		podAutoscalerInternal.UpdateCreationTimestamp(podAutoscaler.CreationTimestamp.Time)
```

Comment:
I suspect that you put that inside `if podAutoscalerInternal.Generation() != podAutoscaler.Generation {`, as we're interested in getting the creation timestamp only once (it won't be only once, but when there's a `.Spec` change, which should not happen frequently)